src/main/java/seedu/mtracker/filemanager/StockDecoder.java

```diff
@@ -0,0 +1,20 @@
+package seedu.mtracker.filemanager;
+
+import seedu.mtracker.model.Instrument;
+import seedu.mtracker.model.InstrumentManager;
+import seedu.mtracker.model.subinstrument.Stock;
+
+public class StockDecoder extends InstrumentDecoder {
+
+    public static final int STOCK_REMARKS_INDEX = 4;
+
+    public static void addStockToList(String[] textSegment, InstrumentManager instrumentManager) {
+        String name = textSegment[NAME_INDEX];
+        double currentPrice = Double.parseDouble(textSegment[CURR_PRICE_INDEX]);
+        String sentiment = textSegment[SENTIMENT_INDEX];
```

Comment:
Can create 3 attributes in the InstrumentDecoder like DECODED_NAME, DECODED_PRICE, DECODED_SENTIMENT,
then use a method called decodeGeneralAttributes to assign these values. Like that for each of the instrument decoder no need to keep repeating these 3 lines